### PR TITLE
Test/testcases target k caculator

### PR DESCRIPTION
**Pull request by hoangdv2429**
Closes: #3436

## What is the purpose of the change

>Add test cases for targetKCaculator

## Brief Changelog

>Add test cases for targetKCaculator in pool-models/stableswap

## Testing and Verifying

This change added tests and can be verified as follows:

add test cases for targetKCaculator, run amm_test for stableswap

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not documented)

### CODE REVIEW

File: x/gamm/pool-models/stableswap/amm_test.go [hunk 1]
```diff
@@ -981,3 +981,65 @@ func TestJoinPoolSharesInternal(t *testing.T) {
 		})
 	}
 }
+
+func TestTargetKCalculator(t *testing.T) {
```
Comment (ValarDragon) [inline x/gamm/pool-models/stableswap/amm_test.go:985]: I don;t understand what this test is actually testing

File: x/gamm/pool-models/stableswap/pool_test.go [hunk 2]
```diff
@@ -1253,3 +1253,41 @@ func TestStableswapSpotPrice(t *testing.T) {
...
+		"number of scaling factors and assets mismatch": {
+			numAssets:      3,
+			scalingFactors: []uint64{10, 10, 10, 10},
+			expectError:    true,
+		},
+		"number of scaling factors is equal to zero": {
+			numAssets:      3,
+			scalingFactors: []uint64{0, 0, 0},
+			expectError:    true,
+		},
```
Comment (ValarDragon) [inline x/gamm/pool-models/stableswap/pool_test.go:1278]:
```suggestion
		"all scaling factors equal to zero": {
			numAssets:      3,
			scalingFactors: []uint64{0, 0, 0},
			expectError:    true,
		},
```

File: x/gamm/pool-models/stableswap/pool_test.go [hunk 3]
```diff
@@ -1253,3 +1253,41 @@ func TestStableswapSpotPrice(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateScalingFactors(t *testing.T) {
```
Comment (ValarDragon) [inline x/gamm/pool-models/stableswap/pool_test.go:1257]: Ty for this test!

File: x/gamm/pool-models/stableswap/amm_test.go (hunk 1 above)
Comment (hoangdv2429) [reply x/gamm/pool-models/stableswap/amm_test.go:985]: As I understand, maybe @p0mvn want to check if given the same conditions. Will the results of calculating be consistent ? as he have said in the issue that I want to close.

File: x/gamm/pool-models/stableswap/pool_test.go (hunk 3 above)
Comment (hoangdv2429) [reply x/gamm/pool-models/stableswap/pool_test.go:1257]: sorry I forgot to checkout  🥲

**Comment by faddat** [comment]
Hey, I did a conflict resolution thing here, you can merge it into your branch :)

https://github.com/osmosis-labs/osmosis/tree/hoangdv/testcases/targetKCalculator

**Comment by ValarDragon** [comment]
Hey, I don't think in its current form this makes sense to merge / effectively tests the function. I'd like to close this for now, along with the original issue

**Pull request closed without merging**